### PR TITLE
search: register repogroup routes in frontend

### PR DESCRIPTION
**Pull request by keegancsmith**
This is to prevent us 404ing when loading a repogroup page. This is
copying the hack in the webapp currently and is not sufficient for when
we roll this out. Instead the repogroups should be living under a
subpath, rather than top-level paths.

Additionally the current logic in the webapp is just a string contains,
so registers many more paths. From testing thankfully our frontend
prefers paths that match repos.

Part of https://github.com/sourcegraph/sourcegraph/issues/11922

### CODE REVIEW

**Comment by keegancsmith** [comment]
I haven't tested this yet! I just want to confirm this route is only registered on paths like `/golang` and not `/golang/.*` before landing.

**Comment by rvantonder** [comment]
Saving you a click on the build output:

```
panic: runtime error: invalid memory address or nil pointer dereference
--
  | [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1784300]
  |  
  | goroutine 1 [running]:
  | github.com/gorilla/mux.(*Route).Handler(...)
  | /root/.asdf/installs/golang/1.14.3/packages/pkg/mod/github.com/gorilla/mux@v1.7.4/route.go:119
  | github.com/sourcegraph/sourcegraph/cmd/frontend/internal/app/ui.initRouter()
  | /buildkite/builds/buildkite-agent-bc856f6b9-clf22-1/sourcegraph/sourcegraph/cmd/frontend/internal/app/ui/router.go:225 +0x21c0
  | github.com/sourcegraph/sourcegraph/cmd/frontend/internal/app/ui.init.1()
  | /buildkite/builds/buildkite-agent-bc856f6b9-clf22-1/sourcegraph/sourcegraph/cmd/frontend/internal/app/ui/router.go:182 +0x30
```

**Comment by attfarhan** [comment]
@keegancsmith thanks for pushing this up. What's left to be done here, just get the build to pass? Would it make sense for me to take over this PR tomorrow?

**Comment by daxmc99** [comment]
Dear all,

This is your release captain speaking. 🚂🚂🚂

Branch cut for the **3.18 release is scheduled for tomorrow**.

Is this issue / PR going to make it in time? Please change the milestone accordingly.
When in doubt, reach out!

Thank you

**Comment by poojaj-tech** [comment]
Yes, @keegancsmith thank you for pushing up this fix!
@attfarhan now that you are back could you help land this PR. I haven't heard from Keegan and we should get this fix out soon.

File: cmd/frontend/internal/app/ui/router.go [hunk 1]
```diff
@@ -134,6 +135,13 @@ func newRouter() *mux.Router {
 	r.PathPrefix("/stats").Methods("GET").Name(routeStats)
 	r.PathPrefix("/views").Methods("GET").Name(routeViews)
 
+	// Repogroup hacks! We don't have a subroute for them at the moment and
+	// the routing is hacked into the webapp code. Before launch this should
+	// be thought through. Copying behaviour from webapp. Must mirror
```
Comment (felixfbecker) [inline cmd/frontend/internal/app/ui/router.go:140]: So when will this be "thought through", and who owns that? Aren't we looking to "launch" this anytime soon?

Comment (attfarhan) [reply cmd/frontend/internal/app/ui/router.go]: I edited this comment. We had a discussion between the product team, Quinn, and myself around having these as top level routes. We're going to go this direction for now, so removed the part about this being a hack and needing more thought.

**Pull request merged**